src/Microsoft.AspNet.SignalR.Client/Transports/AutoTransport.cs

```diff
@@ -74,14 +73,10 @@ public string Name
         public Task<NegotiationResponse> Negotiate(IConnection connection, string connectionData)
         {
             var task = GetNegotiateResponse(connection, connectionData);
-#if NET45
+#if NET45 || NETSTANDARD2_0
```

Comment:
The missing `|| NETSTANDARD2_0` is why clients running on .NET Core could still still connect to the service with WebSockets. .NET Core clients would always try WebSockets regardless of the negotiate response 😞. I checked the rest of the codebase. This is the only place that `#if NET45` appeared by itself without some `|| NETSTANDARD...` component.

Reply:
How about checking if the negotiate response is redirect response or not to determine if `webSockets` should be removed?

Reply:
Is there some reason that's preferable to using the last negotiate response? Because of the layering, I'd rather the AutoTransport not have to determine whether a negotiate response is a redirect or not. It's the code that's calling IClientTransport.Negotiate() that determines this today.

The better layering would be to have the Connection.Negotiate() (which can call the transport's negotiate method multiple times to follow redirects) look at the TryWebSockets value on the last negotiate response and feed the TryWebSockets value back into the transport as a parameter to IClientTransport.Start(). I didn't do that, because I didn't want to have to do a major refactoring and change the IClientTransport interface.